common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -1077,6 +1090,7 @@ public UTF8String translate(Map<Character, Character> dict) {
    * @return true if the parsing was successful else false
    */
   public boolean toLong(LongWrapper toLongResult) {
```

Comment:
let's not make this method has side effect. We should only channe the local variable `offset` inside this method.

Reply:
Do you mean copy this insideTrim logic here?

Reply:
I will make them method local